be/src/vec/functions/array/function_array_distinct.h

```diff
@@ -0,0 +1,294 @@
+
+        DataTypePtr src_column_type = remove_nullable(block.get_by_position(arguments[0]).type);
+        auto nested_type = assert_cast<const DataTypeArray&>(*src_column_type).get_nested_type();
+        auto dest_column_ptr = ColumnArray::create(nested_type->create_column(),
+                                                   ColumnArray::ColumnOffsets::create());
+        IColumn& dest_column = dest_column_ptr->get_data();
+        ColumnArray::Offsets& dest_offsets = dest_column_ptr->get_offsets();
+
+        const IColumn* src_inner_column = nullptr;
+        const ColumnNullable* src_nullable_column =
+                check_and_get_column<ColumnNullable>(src_column_data);
+        if (src_nullable_column) {
+            src_inner_column = src_nullable_column->get_nested_column_ptr();
+        } else {
+            src_inner_column = src_column_array->get_data_ptr();
+        }
+
+        IColumn* dest_nested_column = nullptr;
+        ColumnNullable* dest_nullable_column = reinterpret_cast<ColumnNullable*>(&dest_column);
+        if (dest_nullable_column) {
+            dest_nested_column = dest_nullable_column->get_nested_column_ptr();
+        } else {
+            dest_nested_column = &dest_column;
+        }
+
+        auto res_val =
+                _execute_by_type(*src_inner_column, src_offsets, *dest_nested_column, dest_offsets,
+                                 src_nullable_column, dest_nullable_column, nested_type);
+        if (!res_val) {
+            return Status::RuntimeError(
+                    fmt::format("execute failed or unsupported types for function {}({})",
+                                get_name(), block.get_by_position(arguments[0]).type->get_name()));
+        }
+
+        block.replace_by_position(result, std::move(dest_column_ptr));
+        return Status::OK();
+    }
+
+private:
+    // Note: Here initially allocate a piece of memory for 2^5 = 32 elements.
+    static constexpr size_t INITIAL_SIZE_DEGREE = 5;
+
+    template <typename T>
+    bool _execute_number(const IColumn& src_column, const ColumnArray::Offsets& src_offsets,
+                         IColumn& dest_column, ColumnArray::Offsets& dest_offsets,
+                         const ColumnNullable* src_nullable_col,
+                         ColumnNullable* dest_nullable_col) {
+        const ColumnVector<T>* src_data_concrete =
+                check_and_get_column<ColumnVector<T>>(&src_column);
+        if (!src_data_concrete) {
+            return false;
+        }
+        const PaddedPODArray<T>& src_datas = src_data_concrete->get_data();
+
+        ColumnVector<T>& dest_data_concrete = reinterpret_cast<ColumnVector<T>&>(dest_column);
+        PaddedPODArray<T>& dest_datas = dest_data_concrete.get_data();
+
+        const PaddedPODArray<UInt8>* src_null_map = nullptr;
+        if (src_nullable_col) {
+            src_null_map = &src_nullable_col->get_null_map_column().get_data();
```

Comment:
src_nullable_col->get_null_map_column().get_data().data() returns UInt8*.

use `const UInt8* src_null_map` is simpler.

Reply:
> src_nullable_col->get_null_map_column().get_data().data() returns UInt8*.
> 
> use `const UInt8* src_null_map` is simpler.

this is a good suggestion,  I will simplify here.